qutip/core/superoperator.py

```diff
@@ -96,28 +96,26 @@ def liouvillian(H=None, c_ops=None, data_only=False, chi=None):
                  for c_op, chi_ in zip(c_ops, chi))
         return L
 
-    op_dims = H.dims
-    op_shape = H.shape
-    sop_dims = [[op_dims[0], op_dims[0]], [op_dims[1], op_dims[1]]]
-    sop_shape = [np.prod(op_dims), np.prod(op_dims)]
-    spI = _data.identity(op_shape[0], dtype=type(H.data))
-
-    data = _data.mul(_data.kron(spI, H.data), -1j)
-    data = _data.add(data, _data.kron_transpose(H.data, spI), scale=1j)
+    data = _data.mul(_data.kron(_data.identity_like(H.data), H.data), 
+                     -1j)
+    data = _data.add(data, _data.kron_transpose(H.data,
```

Comment:
I would keep the `spI` definition instead of calling `identity_like` multiple times.